mesonbuild/dependencies/hdf5.py

```diff
@@ -105,7 +105,8 @@ def __init__(self, environment, kwargs):
                 prog = ExternalProgram(wrappers[lang], silent=True)
                 if not prog.found():
                     return
-                cmd = prog.get_command() + ['-show']
+                shlib_arg = '-noshlib' if kwargs.get('static', False) else '-shlib'
+                cmd = prog.get_command() + [shlib_arg + ' -show']
```

Comment:
I think this needs to be, `cmd = prog.get_command() + [shlib_arg', '-show']`, otherwise the quoting will be wrong.

Reply:
@dcbaker thanks for the review. Yes, it should be a list. I just fixed it.

Reply:
Thanks, I'll go ahead and merge this.